Python/pylifecycle.c

```diff
@@ -595,15 +616,32 @@ _Py_SetLocaleFromEnv(int category)
  */
 
 _PyInitError
-_Py_InitializeCore(const _PyCoreConfig *core_config)
+_Py_InitializeCore_impl(PyInterpreterState **interp_p,
+                        const _PyCoreConfig *core_config)
 {
-    assert(core_config != NULL);
-
     PyInterpreterState *interp;
-    PyThreadState *tstate;
-    PyObject *bimod, *sysmod, *pstderr;
     _PyInitError err;
 
+    /* bpo-34008: For backward compatibility reasons, calling Py_Main() after
+       Py_Initialize() ignores the new configuration. */
+    if (_PyRuntime.core_initialized) {
+        PyThreadState *tstate = PyThreadState_GET();
+        if (!tstate) {
+            return _Py_INIT_ERR("failed to read thread state");
+        }
+
+        interp = tstate->interp;
+        if (interp == NULL) {
+            return _Py_INIT_ERR("can't make main interpreter");
```

Comment:
This error message isn't right for the reconfiguration case. Perhaps something like "Core reconfiguration: no main interpreter found".

Then the thread state lookup above could also gain the "Core reconfiguration: " prefix.

Reply:
I changed the error message to "no main interpreter found". The displayed error message contains the name of the current function _Py_InitializeCore_impl() thanks to _Py_INIT_ERR() magic. I don't want to add a prefix per error message. Such error should not occur in practice anyway.